security/pkg/server/monitoring/monitoring.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitoring
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/pprof"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/version"
+)
+
+// Monitor is the server that exposes Prometheus metrics about Citadel.
+type Monitor struct {
+	monitoringServer *http.Server
+	closed           chan bool
+}
+
+const (
+	metricsPath = "/metrics"
+	versionPath = "/version"
+)
+
+// StartMonitor starts the Citadel monitor.
+func StartMonitor(port int, enableProfiling bool, errCh chan<- error) (*Monitor, error) {
```

Comment:
To be consistent, consider to define:
- NewMonitor() *Monitor, error
- Monitor.Start()

Reply:
Done.